### PR TITLE
ui: Implement drawer to add a new team member (PROJQUAY-6032)

### PR DESCRIPTION
**Pull request by harishsurf**
This PR adds the following changes to manage team members view under Teams and membership tab:
- An editable team description section
- A reusable generic delete modal template when deleting a team or a team member
- A drawer to add a member to the team

https://github.com/quay/quay/assets/39300150/0288cad4-6485-4bd1-b829-4b4b7535acfd

### CODE REVIEW

**Review by jonathankingfc — APPROVED** [review]
LGTM

File: web/src/routes/OrganizationsList/Organization/Tabs/TeamsAndMembership/TeamsView/ManageMembers/AddNewTeamMemberDrawer.tsx [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+  DrawerActions,
+  DrawerCloseButton,
+  DrawerHead,
+  DrawerPanelBody,
+  DrawerPanelContent,
+  Form,
+  FormGroup,
+  Spinner,
+} from '@patternfly/react-core';
+import {SelectOption, SelectGroup} from '@patternfly/react-core/deprecated';
```
Comment (jpuzz0) [inline web/src/routes/OrganizationsList/Organization/Tabs/TeamsAndMembership/TeamsView/ManageMembers/AddNewTeamMemberDrawer.tsx:14]: This needs to be updated so both of these components come from `'@patternfly/react-core` instead now.

**Comment by Sunandadadi** [comment]
Great work!

Just one minor comment:
Under `Teamsandmembership` tab,  In team view, when I click on members and delete `user1 ` (or the current logged in user), the API call throws a 404 but I see a success alert - `Successfully deleted team member: user1`. Best if the UI throws whatever the backend is returning.

**Review by Sunandadadi — APPROVED** [review]
lgtm

**Pull request merged**